components/hab/src/config.rs

```diff
@@ -67,7 +67,7 @@ pub fn save(config: &Config) -> Result<()> {
         None => {
             return Err(Error::FileNotFound(
                 config_path.to_string_lossy().into_owned(),
-            ))
+            ));
```

Comment:
As long as we're touching the code, let's simplify it:
```rust
    let parent_path = config_path
        .parent()
        .ok_or_else(|| Error::FileNotFound(config_path.to_string_lossy().into_owned()))?;
```
I've come to recognize that almost any time there's a match statement that does `Some(x) => x` or `Ok(x) => x`, there's a helper that can simplify things.